hummingbot/strategy/strategy_base.pyx

```diff
@@ -130,6 +130,14 @@ cdef class StrategyBase(TimeIterator):
 
         self._sb_order_tracker = OrderTracker()
 
+    def init_params(self, *args, **kwargs):
+        """
+        Assigns strategy parameters, this function must be called directly after init.
+        The reason for this is to make the parameters discoverable through introspect (this is not possible on init of
+        a Cython class).
+        """
+        raise NotImplementedError
+
```

Comment:
In line with our team's decision to include adequate catches for all raised exception, we should instead use the `@abstractmethod` for the `init_params()` and replace `raise NotImplementedError` with a `pass`.

Reply:
Since init_params is not really required to be implemented in the subclass unless the subclass is to be used by HummingSim, we can't enforce abstract method here as this will require any existing strategy, including those by users, to implement it.

Reply:
I think we can enforce this nonetheless since we are defining what needs to be implemented for a strategy to even be considered for code review and possibly merged into our codebase. This also makes it that by default all new incoming strategies would be compatible with HummingSim. 

That being said, since we are enforcing it, we should also abide by the same guidelines and make all existing strategies implement the `init_params()` and make the relevant changes.

Reply:
@CrimsonJacket I'm thinking more of the forked versions of Hummingbot where users create their own strategies, it's unnecessary inconvenient for them to have to update their existing strategies. Also, not all strategies can be back tested with HummingSim. I don't think we should have abstract class and method just for this.

Reply:
Based on @Nullably comment, shouldn't we revisit the discussion about having these configurations that are required for HummingSim made in the client code base?

Reply:
I figure that this should we choose to enforce it, we can schedule a task to make necessary changes. As @Nullably has mentioned, I do agree that there is a level of inconvenience for existing developers but for now I dont see anything wrong with this PR.